pinot-query-planner/src/main/java/org/apache/pinot/calcite/rel/rules/PinotRuleUtils.java

```diff
@@ -46,10 +46,12 @@ private PinotRuleUtils() {
   public static final RelBuilderFactory PINOT_REL_FACTORY =
       RelBuilder.proto(Contexts.of(RelFactories.DEFAULT_STRUCT, PINOT_REL_CONFIG));
 
-  public static final SqlToRelConverter.Config PINOT_SQL_TO_REL_CONFIG =
-      SqlToRelConverter.config().withHintStrategyTable(PinotHintStrategyTable.PINOT_HINT_STRATEGY_TABLE)
-          .withTrimUnusedFields(true).withExpand(true).withInSubQueryThreshold(Integer.MAX_VALUE)
-          .withRelBuilderFactory(PINOT_REL_FACTORY);
+  public static final SqlToRelConverter.Config PINOT_SQL_TO_REL_CONFIG = SqlToRelConverter.config()
+      .withHintStrategyTable(PinotHintStrategyTable.PINOT_HINT_STRATEGY_TABLE)
+      .withTrimUnusedFields(true)
+      .withExpand(true)
+      .withInSubQueryThreshold(0)
```

Comment:
So the idea is to tell calcite to never use INs? Couldn't we just use the default value of 20?

Reply:
I think Jackie's goal is to always convert `IN` to semi-join and `NOT IN` to left-join in the `SqlToRelConverter` and then translate those back to `IN` / `NOT IN` via the new rules? If we use the default value of `20`, we'll need to also add rules to convert the OR'd expressions back to `IN` / `NOT IN` (although they're not particularly problematic if < 20 and we could just let them be I suppose).